mlflow/utils/__init__.py

```diff
@@ -193,3 +193,11 @@ def is_iterator(obj):
     :return: boolean representing whether or not 'obj' is an iterator.
     """
     return (hasattr(obj, "__next__") or hasattr(obj, "next")) and hasattr(obj, "__iter__")
+
+
+def has_ipython():
```

Comment:
I'd rename this function to `is_in_ipython_notebook`. `has_ipython` sounds like we just check whether `IPython` is available.